uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -319,4 +322,326 @@ public function testGetBalanceChangeByAbsenceTypeCanReturnTheBalanceForOnlyPubli
+      'type' => 'All day'
+    ];
+
+    // Wednesday is not a working day on the second week
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-03',
+      'amount' => 0,
+      'type' => 'Non working day'
+    ];
+
+    // Thursday is a working day on the second week
+    $expectedResultsBreakdown['amount'] += 1;
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-04',
+      'amount' => 1.0,
+      'type' => 'All day'
+    ];
+
+    // Friday, Saturday and Sunday are not working days on the second week,
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-05',
+      'amount' => 0,
+      'type' => 'Non working day'
+    ];
+
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-06',
+      'amount' => 0,
+      'type' => 'Weekend'
+    ];
+
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-07',
+      'amount' => 0,
+      'type' => 'Weekend'
+    ];
+
+    // Now, since we hit sunday, the following day will be on the third week
+    // since the start date, but the work pattern only has 2 weeks, so we
+    // rotate back to use the week 1 from the pattern
+
+    // Monday is a working day on the first week
+    $expectedResultsBreakdown['amount'] += 1;
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-08',
+      'amount' => 1.0,
+      'type' => 'All day'
+    ];
+
+    // Tuesday is not a working day on the first week
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-09',
+      'amount' => 0,
+      'type' => 'Non working day'
+    ];
+    // Wednesday is a working day on the first week
+    $expectedResultsBreakdown['amount'] += 1;
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-10',
+      'amount' => 1.0,
+      'type' => 'All day'
+    ];
+    // Thursday is not a working day on the first week
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-11',
+      'amount' => 0,
+      'type' => 'Non working day'
+    ];
+
+    // Friday is a working day on the first week
+    $expectedResultsBreakdown['amount'] += 1;
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-12',
+      'amount' => 1.0,
+      'type' => 'All day'
+    ];
+
+    // Saturday and Sunday are not working days on the first week
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-13',
+      'amount' => 0,
+      'type' => 'Weekend'
+    ];
+
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-14',
+      'amount' => 0,
+      'type' => 'Weekend'
+    ];
+    // Hit sunday again, so we are now on the fourth week since the start date.
+    // The work pattern will rotate and use the week 2
+
+    // Monday is not a working day on week 2
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-08-15',
+      'amount' => 0,
+      'type' => 'Non working day'
+    ];
```

Comment:
also need to use the Option Group to get the 'type' value for all the values on $expectedResultsBreakDown